rust/connlib/snownet/src/node.rs

```diff
@@ -1290,6 +1243,8 @@ struct Connection {
     buffer: Box<[u8; MAX_UDP_SIZE]>,
     intent_sent_at: Instant,
 
+    is_failed: bool,
```

Comment:
Was this added to save CPU time re-checking the timeout repeatedly? As opposed to a `fn is_failed()`, I mean.

Reply:
For the timeout-events, I have no way of checking that I've received the event so I need to set `is_failed` as part of handing it.